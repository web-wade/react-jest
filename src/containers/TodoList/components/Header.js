import React, { Component } from "react";

export default class Header extends Component {
    constructor(props) {
        super(props);
        this.state = {
            value: ""
        };
    }
    render() {
        return (
            <div className="header">
                TodoList
                <input
                    data-test="input"
                    onChange={e => {
                        this.setState({ value: e.target.value });
                    }}
                    value={this.state.value}
                    onKeyUp={this.handInputKeyUp}
                />
            </div>
        );
    }

    handInputKeyUp = e => {
        if (e.keyCode === 13 && this.state.value) {
            this.props.addUndoItem(this.state.value);
            this.setState({
                value:""
            })
        }
    };
}
