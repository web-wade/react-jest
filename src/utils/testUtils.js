export const finTestWrapper = (wrapper,tag) => {
    return wrapper.find(`[data-test='${tag}']`)
}
