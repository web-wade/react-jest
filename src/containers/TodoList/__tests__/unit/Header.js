import React from "react";
import  { shallow } from "enzyme";
import Header from "../../components/Header";
import { finTestWrapper } from "../../../../utils/testUtils";

it("Header 组件包含一个input框", () => {
    const wrapper = shallow(<Header />);
    const inputElem = finTestWrapper(wrapper, "input");
    expect(inputElem.length).toBe(1);
});

it("Heder 组件input框内容，初始化为空", () => {
    const wrapper = shallow(<Header />);
    const inputElem = wrapper.find("[data-test='input']");
    expect(inputElem.prop("value")).toEqual("");
});

it("Heder input用户输入随着改变", () => {
    const wrapper = shallow(<Header />);
    const inputElem = wrapper.find("[data-test='input']");
    inputElem.simulate("change", {
        target: {
            value: "今天要学习Jest"
        }
    });
    expect(wrapper.state("value")).toEqual("今天要学习Jest");
    const newinputElem = wrapper.find("[data-test='input']");
    expect(newinputElem.prop("value")).toEqual("今天要学习Jest");
});

it("Header 组件 inout输入回车时，无内容，无操作", () => {
    const fn = jest.fn();
    const wrapper = shallow(<Header addUndoItem={fn} />);
    const inputElem = wrapper.find("[data-test='input']");
    wrapper.setState({
        value: ""
    });
    inputElem.simulate("keyup", {
        keyCode: 13
    });
    expect(fn).not.toHaveBeenCalled();
});

it("Header 组件 inout输入回车时，有内容，有操作", () => {
    const fn = jest.fn();
    const wrapper = shallow(<Header addUndoItem={fn} />);
    const inputElem = wrapper.find("[data-test='input']");
    wrapper.setState({
        value: "学习React"
    });
    inputElem.simulate("keyup", {
        keyCode: 13
    });
    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenLastCalledWith("学习React");
    expect(inputElem.prop("value")).toBe("");
});

it("Header 组件 inout输入回车时，有内容，有操作,最后应该清除掉", () => {
    const fn = jest.fn();
    const wrapper = shallow(<Header addUndoItem={fn} />);
    const inputElem = wrapper.find("[data-test='input']");
    wrapper.setState({
        value: "学习React"
    });
    inputElem.simulate("keyup", {
        keyCode: 13
    });
    const newinputElem = wrapper.find("[data-test='input']");
    expect(newinputElem.prop("value")).toBe("");
});

it("Header渲染样式正常", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
});
