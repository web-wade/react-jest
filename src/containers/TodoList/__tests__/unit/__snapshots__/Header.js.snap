// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header渲染样式正常 1`] = `
<div
  className="header"
>
  TodoList
  <input
    data-test="input"
    onChange={[Function]}
    onKeyUp={[Function]}
    value=""
  />
</div>
`;
